File: python/sorting/quick_sort_test.py
from .sort import SortType
from .quick_sort import QuickSort
from .perf_measurer import PerfMeasurer


def test_quick_sort():
    arr = [76, 34, 67, 32, 78, 45]
    sorter = QuickSort()
    sorter.sort(arr)
    expected_asc = [32, 34, 45, 67, 76, 78]
    assert arr == expected_asc

    sorter.sort(arr, SortType.desc)
    expected_desc = [78, 76, 67, 45, 34, 32]
    assert arr == expected_desc

    perfMeasurer = PerfMeasurer(sorter)
    print(perfMeasurer.reportPerf())

File: go/sorting/quick_sort_test.go
package sorting

import (
	"fmt"
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestQuickSort(t *testing.T) {
	arr := []int{76, 34, 67, 32, 78, 45}
	sorter := QuickSort{}

	sorted_asc := sorter.Sort(arr, "asc")
	expected_asc := []int{32, 34, 45, 67, 76, 78}
	assert.Equal(t, expected_asc, sorted_asc)

	perfMeasurer := PerfMeasurer{sorter: sorter}
	fmt.Println(perfMeasurer.ReportPerf())

}
